### PR TITLE
[Feature]: Email verification login flow

### PR DESCRIPTION
**Pull request by Anish-Agnihotri**
Implements a custom verification page (post-login request) and verification email.

### Verification page
Adds a page under `/pages/verify.js` that contains the post-login request verification page. In order to include the requested email in the message, the email is stored in `localStorage` during the initial request on the login page. An abstraction for this via hooks has been included in `utils/useLocalStorage.js`.

### Verification email
1. Adds a `sendVerificationRequest` handler to the next-auth email provider.
2. Implements a `nodemailer` based custom email (in both `html` and `text` fallback variants).

## Screenshots
**Custom page**:

Screen Shot 2021-03-06 at 3 29 25 PM

**Custom email**:
Screen Shot 2021-03-06 at 3 29 57 PM

## Merge blockers
Until this change is merged into `master`, the Dancefest logo in the html email template will not display because it is hardcoded to `https://www.ossdancefest.com/images/logo.png`. Not a merge blocker perse, but something to make note of.

### CODE REVIEW

**Review by thepenguinco — APPROVED** [review]
LGTM.  Let's just get one more review on this, then it can go in.

File: utils/sendVerificationRequest.js [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+ * @returns promise, resolving to a successful nodemailer email send
+ */
+const sendVerificationRequest = ({ identifier: email, url, provider }) => {
+  // Return a new promise
+  return new Promise((resolve, reject) => {
+    // Collect mail server and from email
+    const { server, from } = provider;
+
+    // Create mail with server
+    nodemailer.createTransport(server).sendMail(
```
Comment (feng-eric) [inline utils/sendVerificationRequest.js:17]: This code seems to be the same as the code in aws/index.js. Perhaps you could just import that file instead?

**Review by feng-eric — APPROVED** [review]
Other than that one comment I had, it looks good to me

**Comment by mkanoria** [comment]
LGTM. Just one small clarification, the file ` public/images/logo.png` seems to be empty. Is it supposed to be a symlink?

**Review by mkanoria — APPROVED** [review]

**Pull request merged**